### ISSUE TITLE
i18next option 'lng' and 0.51.0 version crashe the nextjs app

### ISSUE THREAD

**Issue by lavir** [isaachinman/next-i18next#418]
## Describe the bug
After updating the next-i18next to 0.51.0 our app crashes. 

The error:

`Error: No router instance found. you should only use "next/router" inside the client side of your app. https://err.sh/zeit/next.js/no-router-instance`

I think that the issue is here:

```
File: src/hocs/app-with-translation.js
27:       i18n.on('languageChanged', (lng) => {
28:         const { router } = props
29:         const { pathname, asPath, query } = router
30:         const routeInfo = { pathname, query }
31: 
32:         if (process.browser && config.localeSubpaths !== localeSubpathOptions.NONE) {
33:           const { as, href } = lngPathCorrector(config, { as: asPath, href: routeInfo }, lng)
34:           if (as !== asPath) {
35:             router.replace(href, as)
36:           }
37:         } else {
38:           router.replace({ pathname, query })   //  <<<<<
39:         }
40:       })
```

## Occurs in next-i18next version
version 0.51.0

## Steps to reproduce
1. Open the app 'examples/simple' in next-i18next. 
2. Then add the option `lng: 'en'` in i18.js file.
3. Rerun the server.

## Expected behaviour
At least it should not crash

**Comment by isaachinman** [comment]
That's absolutely correct - my apologies. This has been fixed with #419, and I have written test coverage to specifically cover this case to prevent regressions in the future.

@lavir If you'd like to review #419 I'd appreciate it. I can merge and release this ASAP.

**Comment by lavir** [comment]
Thank you! 
I will check PR now

**Comment by isaachinman** [comment]
@lavir Released in v0.51.1.
